src/Adapter/ParameterContainer.php

```diff
@@ -64,37 +64,37 @@ class ParameterContainer implements Iterator, ArrayAccess, Countable
      */
     public function __construct(array $data = [])
     {
-        if ($data) {
+        if ($data !== []) {
             $this->setFromArray($data);
         }
     }
 
     /**
      * Offset exists
      *
-     * @param  string $name
+     * @param  string $offset
      * @return bool
      */
-    #[ReturnTypeWillChange]
-    public function offsetExists($name)
+    #[Override] #[ReturnTypeWillChange]
+    public function offsetExists($offset)
     {
-        return isset($this->data[$name]);
+        return isset($this->data[$offset]);
     }
 
     /**
      * Offset get
      *
-     * @param  string $name
+     * @param  string $offset
      * @return mixed
      */
-    #[ReturnTypeWillChange]
-    public function offsetGet($name)
+    #[Override] #[ReturnTypeWillChange]
+    public function offsetGet($offset)
```

Comment:
Why the variable change here but not in other places? For instance it is still $name in offsetSetReference. If you're change it then change it everywhere. Since many other $variables have contextual names that are descriptive of the state of the data I think $name here is actually better since $normalizedName is a reference to $name after normalization. If you are going to change $name to $offset then it should then be $normalizedOffset.